numpy/core/src/multiarray/buffer.c

```diff
@@ -509,6 +509,10 @@ _buffer_info_new(PyObject *obj)
     PyArray_Descr *descr = NULL;
     int err = 0;
 
+    /*
+     * Note that the buffer info is cached as pyints making them appear like
+     * unreachable lost memory to valgrind.
+     */
```

Comment:
Frankly, right now I do not understand the logic of the cache, but I did not try to either and I guess they are very small objects.